File: from-0101-to-0200/0129-sum-root-to-leaf-numbers/solution.go
/**
 * Definition for a binary tree node.
 * type TreeNode struct {
 *     Val int
 *     Left *TreeNode
 *     Right *TreeNode
 * }
 */
 func sumNumbers(root *TreeNode) int {
    result := 0
    var dfs func(node *TreeNode, curr int)
    dfs = func(node *TreeNode, curr int) {
        if node == nil {
            return
        }
        curr = curr * 10 + node.Val
        if node.Left == nil && node.Right == nil {
            result += curr
        }
        dfs(node.Left, curr)
        dfs(node.Right, curr)
    }
    dfs(root, 0)
    return result
}

File: from-0101-to-0200/0129-sum-root-to-leaf-numbers/solution.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
class Solution:
    def sumNumbers(self, root: Optional[TreeNode]) -> int:
        result = 0

        def dfs(node: TreeNode, curr: int) -> None:
            if not node:
                return

            nonlocal result
            curr = curr * 10 + node.val
            if not node.left and not node.right:
                result += curr
                return

            dfs(node.left, curr)
            dfs(node.right, curr)
            return

        dfs(root, 0)
        return result
        